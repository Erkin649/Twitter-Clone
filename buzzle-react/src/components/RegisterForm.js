import React from 'react';
import { Redirect } from "react-router";

class RegisterForm extends React.Component {
    componentWillUnmount() {
        this.props.unloadRegisterForm();
        this.props.clearError();
    }

    handleChange = event => {
        this.props.updateRegisterForm(event.target.name, event.target.value);
    }

    handleSubmit = event => {
        event.preventDefault();

        this.props.createUser({
            'login': this.props.login,
            'password': this.props.password,
            'password-repeat': this.props['password-repeat']
        });
    }

    render() {
        const user = this.props.user;
        const login = this.props.login;
        const password = this.props.password;
        const passwordRepeat = this.props['password-repeat'];

        return (user ?
            <Redirect to="/" />
            :
            <div className="row justify-content-center">
                <div className="col-lg-6 px-5">
                    <form
                        className="p-3"
                        method="POST"
                        action="/register"
                        autoComplete="off"
                        onSubmit={this.handleSubmit} >
                        <div className="form-group">
                            <label htmlFor="login">Login:</label>
                            <input
                                id="login"
                                className="form-control"
                                type="text"
                                name="login"
                                autoComplete="off"
                                value={login}
                                onChange={this.handleChange}
                                required />
                        </div>
                        <div className="form-group">
                            <label htmlFor="password">Password:</label>
                            <input
                                id="password"
                                className="form-control"
                                type="password"
                                name="password"
                                autoComplete="new-password"
                                onChange={this.handleChange}
                                value={password}
                                required />
                        </div>
                        <div className="form-group">
                            <label htmlFor="password-repeat">Repeat the Password:</label>
                            <input
                                id="password-repeat"
                                className="form-control"
                                type="password"
                                name="password-repeat"
                                autoComplete="new-password"
                                onChange={this.handleChange}
                                value={passwordRepeat}
                                required />
                        </div>
                        <input className="btn btn-primary" type="submit" value="Register" />
                    </form>        
                </div>
            </div>
        );
    }
}

export default RegisterForm;
