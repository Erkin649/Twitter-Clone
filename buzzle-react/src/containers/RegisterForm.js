import { connect } from 'react-redux';

import { clearError } from '../actions/error.js';
import { createUser } from '../actions/user.js';
import { updateRegisterForm, unloadRegisterForm } from '../actions/forms.js';
import RegisterForm from '../components/RegisterForm.js';

const mapStateToProps = state => ({
    'user': state.user,
    'login': state.registerForm.login,
    'password': state.registerForm.password,
    'password-repeat': state.registerForm['password-repeat']
});

const mapDispatchToProps = dispatch => ({
    'clearError': () => dispatch(clearError()),
    'createUser': credentials => dispatch(createUser(credentials)),
    'updateRegisterForm': (name, value) => dispatch(updateRegisterForm(name, value)),
    'unloadRegisterForm': () => dispatch(unloadRegisterForm())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(RegisterForm);
