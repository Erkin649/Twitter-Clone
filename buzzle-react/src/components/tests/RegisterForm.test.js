import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import RegisterForm from '../RegisterForm.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RegisterForm />, div);
});

it('renders correctly and equal to snapshot', () => {
    const tree = renderer.create(<RegisterForm />)
                         .toJSON();

    expect(tree).toMatchSnapshot();
});