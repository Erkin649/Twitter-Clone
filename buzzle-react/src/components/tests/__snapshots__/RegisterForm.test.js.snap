// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and equal to snapshot 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-lg-6 px-5"
  >
    <form
      action="/register"
      autoComplete="off"
      className="p-3"
      method="POST"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="login"
        >
          Login:
        </label>
        <input
          autoComplete="off"
          className="form-control"
          id="login"
          name="login"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <input
          autoComplete="new-password"
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password-repeat"
        >
          Repeat the Password:
        </label>
        <input
          autoComplete="new-password"
          className="form-control"
          id="password-repeat"
          name="password-repeat"
          onChange={[Function]}
          required={true}
          type="password"
        />
      </div>
      <input
        className="btn btn-primary"
        type="submit"
        value="Register"
      />
    </form>
  </div>
</div>
`;
